lib/dukpt.lib.js

```diff
@@ -2,10 +2,10 @@ const aesjs = require('aes-js');
 const DataOperations = require('./data.lib');
 
 class Dukpt {
-    constructor(bdk, ksn) {
+    constructor(bdk, ksn, keymode = 'datakey') {
         this.bdk = bdk;
         this.ksn = ksn;
-        this._sessionKey = this._deriveDukptSessionKey();
+        this._sessionKey = this._deriveDukptSessionKey(datakey);
```

Comment:
did you look at just the first commit? `datakey` was renamed to `keyMode` in my last commit.